cycler/__init__.py

```diff
@@ -132,14 +144,21 @@ class Cycler:
     def __call__(self):
         return cycle(self)
 
-    def __init__(self, left, right=None, op=None):
+    def __init__(
+        self,
+        left: Cycler[K, V] | Iterable[dict[K, V]] | None,
+        right: Cycler[K, V] | Iterable[dict[K, V]] | None = None,
+        op: Any = None,
```

Comment:
slightly lazy to use `Any` here, but this is pretty much zip|np.product|None, exactly but I could not find a good way to type hint that (tried using Literals, but those didn't work with functions and are not available on all supported python versions anyway)